Add porting guide for HAL I2C API

maciejbocianski opened the pull request:
This is updated version of #807

changes compared to #807:
- sync with I2C [API](https://github.com/ARMmbed/mbed-os/blob/feature-i2c/hal/i2c_api.h) and [docs](https://github.com/ARMmbed/mbed-os/blob/feature-i2c/docs/design-documents/hal/0001-i2c-overhaul.md) changes
- change links to point to feature-i2c branch as development is on feature branch
- change test command to new I2C fpga test (there is no non-fpga test for I2C) (https://github.com/ARMmbed/mbed-os/tree/feature-i2c/TESTS/mbed_hal_fpga_ci_test_shield/i2c)

I don't know how to fix following links:
http://os-doc-builder.test.mbed.com/docs/development/feature-i2c-doxy/classmbed_1_1_i_2_c.html
http://os-doc-builder.test.mbed.com/docs/development/feature-hal-i2c-doxy/group__hal__i2c__tests.html


@donatieng @jamesbeyond @MarceloSalazar @ashok-rao @AnotherButler 
please review

maciejbocianski committed:
Add porting guide for HAL I2C API

Commit:
Edit i2c.md

Edit file, mostly for consistent tense and international spelling.

Commit:
Fix links in i2c.md

Fix outdated links that don't work anymore.

maciejbocianski committed:
i2c porting guide: add i2c_set_clock_stretching

maciejbocianski commented:
`i2c_set_clock_stretching` added, @AnotherButler please re-review

AnotherButler reviewed (approved):
Nice work 👍 Thanks for the PR.

AnotherButler commented:
ping @donatieng @jamesbeyond @MarceloSalazar @ashok-rao

AnotherButler commented:
ping @jamesbeyond @MarceloSalazar @ashok-rao @donatieng

jamesbeyond reviewed (changes requested):
shall we added the warnings about: this new API is on a feature branch, not master.
just like: https://github.com/ARMmbed/mbed-os-5-docs/blob/development/docs/porting/target/spi.md#serial-peripheral-interface-spi
 
Otherwise it all looks good

Commit:
Add warning to i2c.md

Add warning about code being on a feature branch to match docs in the rest of the section.

ashok-rao reviewed (approved):
LGTM! Thanks!

AnotherButler commented:
@jamesbeyond Is this OK now?

jamesbeyond reviewed (approved):
Looks good , Thanks @AnotherButler
